### PR TITLE
Add encryption API reference

### PR DESCRIPTION
**Pull request by EugeniyBurmistrov**

### CODE REVIEW

**Review by const314 — APPROVED** [review]

File: docs/api/encryption/factory-function.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+---
+id: factory-function
+title: Factory Function
+---
+
+# Encryption Factory Function
+
+## Overview
+
+To implement aggregate or read model encryption, you need to define an encryption factory function. This function receives data about the currently processed operation and returns an object that contains your implementation of an `encrypt` and `decrypt` functions.
```
Comment (albertov05) [inline docs/api/encryption/factory-function.md:10]:
```suggestion
To implement aggregate or read model encryption, you need to define an encryption factory function. This function receives data about the currently processed operation and returns an object that contains your implementation of `encrypt` and `decrypt` functions.
```

File: docs/api/encryption/secrets-manager.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+| ------------- | -------- | ------------------------------------------------------ |
+| `id`          | `string` | The secret's unique identifier within the event store. |
+
+**Result**
+
+A promise that resolves to either the loaded secret or `null` if the secret with the specified `id` was not found.
+
+## `setSecret`
+
+Saves the specified secrete to the event store.
```
Comment (albertov05) [inline docs/api/encryption/secrets-manager.md:64]:
```suggestion
Saves the specified secret to the event store.
```

File: docs/encryption.md [hunk 3]
```diff
@@ -90,13 +90,15 @@ const createEncryption = (aggregateId, context) => {
...
-| `getSecret`    | Takes a unique ID as an argument and returns a promise that resolves to a string if a secret was found or null if a secret was not found. |
-| `setSecret`    | Takes a unique ID and a secret string as arguments and returns a promise that resolves if the secret was successfully saved.              |
-| `deleteSecret` | Takes a unique ID as an argument and returns a promise that resolves if the secret was successfully deleted.                              |
+| Function Name                                                    | Description                                                                                                                               |
+| ---------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------- |
+| [`getSecret`](api/encryption/secrets-manager.md#getsecret)       | Takes a unique ID as an argument and returns a promise that resolves to a string if a secret was found or null if a secret was not found. |
```
[the remark this answers is not in the document]
Comment (EugeniyBurmistrov) [reply docs/encryption.md:95]: Yes, it should be in quotes, thank you

File: docs/api/encryption/secrets-manager.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+**Arguments**
+
+| Argument Name | Type     | Description                                            |
+| ------------- | -------- | ------------------------------------------------------ |
+| `id`          | `string` | The secret's unique identifier within the event store. |
+
+**Result**
+
+A promise that resolves to a `booolean` value. The value indicates whether or not a secret with the specified `id` has been found and successfully deleted.
```
[the remark this answers is not in the document]
Comment (EugeniyBurmistrov) [reply docs/api/encryption/secrets-manager.md]: Fixed the typo.

> does the first letter need to be capitalized?

No, it is a primitive type, so the lower case is correct.

File: docs/encryption.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (albertov05) [reply docs/encryption.md]: Thank you.

File: docs/api/encryption/secrets-manager.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (albertov05) [reply docs/api/encryption/secrets-manager.md]: Thank you.

**Pull request merged**